improver/cli/estimate_emos_coefficients_from_table.py

```diff
@@ -153,15 +153,22 @@ def process(
         EstimateCoefficientsForEnsembleCalibration,
     )
 
-    filters = [("diagnostic", "==", diagnostic)]
+    # Load forecasts from parquet file filtering by diagnostic and blend_time.
+    forecast_period_td = pd.Timedelta(int(forecast_period), unit="seconds")
+    # tz_localize(None) is used to facilitate filtering, although the dataframe
+    # is expected to be timezone aware upon load.
+    cycletimes = pd.date_range(
+        end=pd.Timestamp(cycletime)
+        - pd.Timedelta(1, unit="days")
+        - forecast_period_td.floor("D"),
+        periods=int(training_length),
+        freq="D",
+    ).tz_localize(None)
+    filters = [[("diagnostic", "==", diagnostic), ("blend_time", "in", cycletimes)]]
```

Comment:
It would make sense to add any further filtering needed (eg. filter on experiment) here also. Even if it won't help with the current form of the data